### DISCUSSION TITLE
v7.2.0

### DISCUSSION THREAD

**Discussion by Taiko2k** [Taiko2k/Tauon#749] [General]
## Welcome to v7.2.0

This one has taken a little while. But with the many subtle fixes and improvements, Tauon is really starting to become a polished app!

#### On becoming a supporter

It can be pretty painful to donate to your favourite projects when you don't feel like you're directly getting anything for it. Our R&D team has been hard at work finding a way to get you something extra without flying in the face of free software ideals. With that in mind, we have a result, a new supporters exclusive theme!

## Introducing a new theme: Charmed (***supporter exclusive*** :sparkling_heart:)

***Charmed*** is a gorgeous new deep red theme featuring *Rizumu* the witch, your new magical music companion! :sparkle: 

![Screenshot from 2022-05-16 15-26-52](https://user-images.githubusercontent.com/17271572/168514949-13825b22-0376-481d-8c9d-30614e9a4dbe.png)

*Will you let her charm her way into your heart?* 

This theme is available for sponsors on tier 2 or higher. If you'd like to join, see my GitHub sponsors page [here](https://github.com/sponsors/Taiko2k).

## Gallery singles thumbnail style

![Screenshot from 2022-04-13 19-34-29](https://user-images.githubusercontent.com/17271572/164357799-dc6613c3-f44d-48fe-bf82-d3252be613b9.png)

Folders of singles now have a new stacked thumbnail style in the gallery. This is a little experimental and may need some further tweaking. Let me know what you think.

## Removal of GStreamer backend

Big change with this release is the removal of the GStreamer backend. The new Phazor backend is about 2 years old now and in that time has shaved a lot of its rough edges. Its still not perfect, but at this point, I have lost interest in maintaining the GStreamer backend. 

This means that the Phazor backend is now the default and only backend. The Phazor branding in the UI has been removed, so it can just be considered part of Tauon without need of distinction.

## Other changes

 - Improved behaviour of (jump) track crossfade. It now fades out immediately and fixes an issue with stuttering on IO delay.
 - There are now settings for `samplerate-output` and `resample-quality`. (For the average user there's probably no need to change these.)
 - There is a new setting `volume-curve` in the config file. It still uses non-linear by default, but you can now change it to linear.
 - Added native Wavpack decoding. (Using the correction file is now defined behaviour.)
 - Added new generator code `px` to exclude playlists of tracks.
 - Restored Spotify feature support. (Due to consideration from user feedback.)
 - Fixed delay/freeze on changing audio output device.
 - Mini modes are now launch-able from *shuffle-lockdown* mode.
 - Themes can now be organised into subfolders.
 - You can now use <kbd>Ctrl</kbd> + <kbd>m</kbd> to toggle mute.
 - There's now an 'auto` mode for Replay Gain that automatically switches between track and album gain depending on shuffle mode.
 - Various other fixes and tweaks.

## Network streaming cache system

This update brings a reworked network file cache system. This rework brings the following changes:

 - Implemented a pre-emptive caching mechanism that should improve gapless playback reliability.
 - You can now enable caching for *local* files. This is useful for playing from SMB/NFS mounts on slower connections.
 - There's now an option to **persistently** store cached tracks up to a given limit. Using this may improve app performance, particularly if you often repeat playing the same albums.

If you use streaming in Tauon, let me know what you think of this feature.
 
## [Flatpak] Codec support changes

Support for the codecs **WMA** and **APE** is no longer included with the base Flatpak install of Tauon. These codecs now require an external optional add-on package. To install this just run:

  >  flatpak install org.freedesktop.Platform.ffmpeg-full//21.08

Codec support for **TTA** has been lost. Let me know if anyone used this codec.

## [AUR] JPEG XL support

Experimental JPEG XL support has been added to make it possible to read album art in this format. For this you'll need to install the new optional dependency `python-jxlpy` ([AUR](https://aur.archlinux.org/packages/python-jxlpy)). Not yet included with Flatpak install.

<hr /><em>This discussion was created from the release <a href='https://github.com/Taiko2k/TauonMusicBox/releases/tag/v7.2.0'>v7.2.0</a>.</em>

**Comment by Solarunit** [comment]
>You can now use Ctrl + m to toggle mute.

Thanks!

Also new theme looks nice.

**Comment by Solarunit** [comment]
Could we have some indication when Tauon in the mute state?

![alt text](https://i.imgur.com/o4T0raD.png)

Maybe an icon in the volume bar like this?

![alt text](https://i.imgur.com/kk9ESB7.png)
